### PR TITLE
Prometheus 2.40.2

### PR DESCRIPTION
**Pull request by lomik**
Upgrade built-in prometheus engine and ui to latest 2.40.2 version

Breaking changes:
The prometheus listener is now separate from the main http listener.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -290,6 +291,7 @@ func New() *Config {
 		Prometheus: Prometheus{
 			ExternalURLRaw: "",
 			PageTitle:      "Prometheus Time Series Collection and Processing Server",
+			Listen:         ":9090",
```
Comment (msaf1980) [inline config/config.go:294]: Possibly conflict with main http listener ?

Comment (lomik) [reply config/config.go]: yep. changed to 9092

File: config/config_test.go [hunk 2]
```diff
@@ -383,7 +383,7 @@ sample-thereafter = 12
 	assert.Equal(t, expected.Carbonlink, config.Carbonlink)
 
 	// Prometheus
-	expected.Prometheus = Prometheus{"https://server:3456/uri", nil, "Prometheus Time Series"}
+	expected.Prometheus = Prometheus{":9090", "https://server:3456/uri", nil, "Prometheus Time Series"}
```
Comment (msaf1980) [inline config/config_test.go:386]: And this need to be changed to :9092.

File: config/config_test.go [hunk 3]
```diff
@@ -622,7 +622,7 @@ sample-thereafter = 12
...
```
Comment (msaf1980) [inline config/config_test.go:625]: And this need to be changed to :9092.

**Pull request merged**